vm/src/format.rs

```diff
@@ -155,16 +155,19 @@ fn parse_fill_and_align(text: &str) -> (Option<char>, Option<FormatAlign>, &str)
     }
 }
 
-fn parse_number(text: &str) -> (Option<usize>, &str) {
+fn parse_number(text: &str) -> Result<(Option<usize>, &str), &'static str> {
     let num_digits: usize = get_num_digits(text);
     if num_digits == 0 {
-        return (None, text);
+        return Ok((None, text));
     }
-    // This should never fail
-    (
+    if num_digits >= 20 {
+        // NOTE: this is different from CPython
+        return Err("Too many decimal digits in format string");
+    }
+    Ok((
         Some(text[..num_digits].parse::<usize>().unwrap()),
```

Comment:
Why not checking the return value of `parse` instead of checking for number of digits?